process/scToProtocol/stakingToPeer.go

```diff
@@ -187,15 +188,19 @@ func (stp *stakingToPeer) updatePeerState(
 		return err
 	}
 
-	isValidator := account.GetList() == string(core.EligibleList) || account.GetList() == string(core.WaitingList)
-	unJailForInactive := len(account.GetBLSPublicKey()) > 0 && stakingData.UnJailedNonce == nonce && !isValidator
-	if unJailForInactive {
+	isUnJailForInactive := len(account.GetBLSPublicKey()) > 0 && !stakingData.Staked &&
```

Comment:
If node is jailed for inactivity then stakingData.Staked is false ?

Reply:
It depends. If there was a switch with a node in waiting list or not. If a switch was done, than the node is unStaked as well, thus stake == false

Reply:
Otherwise it is still staked

Reply:
ok